src/preferences/dialog/dlgprefwaveform.cpp

```diff
@@ -244,7 +248,8 @@ void DlgPrefWaveform::slotSetWaveformType(int index) {
     if (index < 0) {
         return;
     }
-    WaveformWidgetFactory::instance()->setWidgetTypeFromHandle(index);
+    int handleIndex = waveformTypeComboBox->itemData(index).toInt();
```

Comment:
Casting from int to enum should work.

Reply:
yes, it does.
for the sake of completeness: unfortunately we can not use the type enum without a large refactoring of WaveformWidgetFactory in order to update the widgets immediately.
explananed in https://github.com/mixxxdj/mixxx/pull/3980#discussion_r650529652